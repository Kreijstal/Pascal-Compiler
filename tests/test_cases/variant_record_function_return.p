program minimal_record;

type
  TTestType = (ttOne, ttTwo);

  TTestRecord = record
    case ValueType: TTestType of
      ttOne: (IntValue: Integer);
      ttTwo: (StrValue: string);
  end;

var
  rec: TTestRecord;

function TestFunction: TTestRecord;
begin
  TestFunction.ValueType := ttOne;
  TestFunction.IntValue := 42;
end;

begin
  rec.ValueType := ttOne;
  rec.IntValue := 10;
  WriteLn('Record test');
  rec := TestFunction;
  WriteLn('Function result: ', rec.IntValue);
end.
