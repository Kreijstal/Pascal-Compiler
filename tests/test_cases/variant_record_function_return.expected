Record test
Function result: 42
